### DISCUSSION TITLE
Bug report regarding string.replace() inside a formatter line

### DISCUSSION THREAD

**Discussion by rmetrich** [micropython/micropython#13401] [Core Development]
Hi,
I've started using micropython on a esp8266, it's just great!
I would like to report a bug in string's replace() implementation with colon replacement.
I could confirm it also fails with the online unicorn emulator, please find below the code excerpt:
~~~
>>> temps = dict()
>>> temps['a'] = "20.5:1"
>>> "&".join(map(lambda i: f"{i[0]}={i[1].replace(':', 'foo')}", temps.items()))
Traceback (most recent call last):
  File "<stdin>", line 1
SyntaxError: invalid syntax
~~~

This kind of code works find on regular python.
It seems to be an issue only when using a `f""` formatter.

**Comment by scruss** [comment]
There are a lot of f-string features that MicroPython doesn't support: [MicroPython differences from CPython | Core language](https://docs.micropython.org/en/latest/genrst/core_language.html#core-language).

There must be a less convoluted way of expressing this code.

**Reply by mendenm** [reply]
One of the specifically unavailable features is object indexing inside an f-string.
